packages/millicast-sdk/src/PeerConnectionStats.js

```diff
@@ -62,221 +63,62 @@ export const peerConnectionStatsEvents = {
 }
 
 export default class PeerConnectionStats extends EventEmitter {
-  constructor (peer) {
+  constructor (peer, options = { autoInitStats: true }) {
     super()
     this.peer = peer
-    this.stats = null
-    this.emitInterval = null
-    this.previousStats = null
+    this.collection = null
+    this.initialized = false
+
+    if (options.autoInitStats) {
+      this.init()
+    }
   }
 
   /**
    * Initialize the statistics monitoring of the RTCPeerConnection.
    */
   init () {
+    if (this.initialized) {
+      logger.warn('PeerConnectionStats.init() has already been called. Automatic initialization occurs when the PeerConnectionStats object is constructed.')
+      return
+    }
     logger.info('Initializing peer connection stats')
-    this.emitInterval = setInterval(async () => {
-      const stats = await this.peer.getStats()
-      this.parseStats(stats)
-      Diagnostics.addStats(this.stats)
-      /**
-       * Peer connection incoming stats.
-       *
-       * @event PeerConnection#stats
-       * @type {ConnectionStats}
-      */
-      this.emit(peerConnectionStatsEvents.stats, this.stats)
-    }, 1000)
+    const peer = this.peer
+    try {
+      this.collection = new WebRTCStats({
+        getStatsInterval: 1000,
+        getStats: () => {
+          return peer.getStats()
+        },
+        includeRawStats: true
+      })
+      this.collection.on('stats', (stats) => {
+        Diagnostics.addStats(stats)
+        this.emit(peerConnectionStatsEvents.stats, stats)
+      })
+      this.collection.start()
+    } catch (e) {
+      logger.error(e)
+    }
+    this.initialized = true
   }
 
   /**
    * Parse incoming RTCPeerConnection stats.
+   * @deprecated since version 0.1.45 - will be removed in future releases.
    * @param {RTCStatsReport} rawStats - RTCPeerConnection stats.
-   * @returns {ConnectionStats} RTCPeerConnection stats parsed.
+   * @returns {null} Method deprecated and no longer returns meaningful data.
    */
   parseStats (rawStats) {
```

Comment:
For customers who use the SDK via CDN and not NPM, this might result in a breaking change. 

I like the warning, but we might have to keep this method around for a bit. 

@j12y @vincentsong what do you guys think?

Reply:
I may need some clarification on scope to answer, because if this is only used in the results of `diagnose()` then that is fine, we want `diagnose()` to return webrtc-stats library results and keys.

If this is a change to both `diagnose()` and the results of `webRTCPeer.on('stats', ...)` then that is more than we want to change at this time. There would be dependencies on hosted player, etc. we would need to reconcile for that deep of a change.

in other words:
- if diagnose is all that is impacted, this is fine
- if all `initStats()` users are impacted, then that's a bigger deal we should revisit

Reply:
@dubeyShivank It looks like this method never returned an object like that. 

It is used to parse the raw stats in the original interval and was exposed publicly.

I think it will be safe to deprecate it.

Reply:
Ok sweet - thanks for confirming!